test/widgets/login_test.dart

```diff
@@ -118,30 +123,71 @@ void main() {
           });
       }
 
-      testWidgets('basic happy case', (tester) async {
-        final serverSettings = eg.serverSettings();
-        await prepare(tester, serverSettings);
-        check(testBinding.globalStore.accounts).isEmpty();
-
-        await tester.enterText(findUsernameInput, eg.selfAccount.email);
+      Future<void> login(WidgetTester tester, Account account) async {
+        await tester.enterText(findUsernameInput, account.email);
         await tester.enterText(findPasswordInput, 'p455w0rd');
         testBinding.globalStore.useCachedApiConnections = true;
         connection.prepare(json: FetchApiKeyResult(
-          apiKey: eg.selfAccount.apiKey,
-          email: eg.selfAccount.email,
-          userId: eg.selfAccount.userId,
+          apiKey: account.apiKey,
+          email: account.email,
+          userId: account.userId,
         ).toJson());
         await tester.tap(findSubmitButton);
-        checkFetchApiKey(username: eg.selfAccount.email, password: 'p455w0rd');
+        checkFetchApiKey(username: account.email, password: 'p455w0rd');
         await tester.idle();
+      }
+
+      testWidgets('basic happy case', (tester) async {
+        final serverSettings = eg.serverSettings();
+        await prepare(tester, serverSettings);
+        checkNotLoggedInStartingRoutes();
+        pushedRoutes.clear();
+        check(testBinding.globalStore.accounts).isEmpty();
+
+        await login(tester, eg.selfAccount);
         check(testBinding.globalStore.accounts).single
           .equals(eg.selfAccount.copyWith(
             id: testBinding.globalStore.accounts.single.id));
       });
 
+      testWidgets('logging into a different account', (tester) async {
+        await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+        final serverSettings = eg.serverSettings();
+        await prepare(tester, serverSettings);
+        check(poppedRoutes).isEmpty();
+        check(pushedRoutes).deepEquals(<Condition<Object?>>[
+          (it) => it.isA<MaterialAccountWidgetRoute>()
+            ..accountId.equals(eg.selfAccount.id)
+            ..page.isA<HomePage>(),
+          (it) => it.isA<MaterialWidgetRoute>()
+            ..page.isA<LoginPage>(),
+        ]);
+        pushedRoutes.clear();
+
+        await login(tester, eg.otherAccount);
+        final loggedInAccount = testBinding.globalStore.accounts.singleWhere(
```

Comment:
nit: "logged-in account" is a confusing name here, because both accounts are logged in.

```suggestion
        final newAccount = testBinding.globalStore.accounts.singleWhere(
```